libs/category/src/effects/category-page.effects.spec.ts

```diff
@@ -0,0 +1,369 @@
+        actions$ = hot('--a', { a: categoryLoadAction });
+        expected = cold('--b', { b: categoryLoadFailureAction });
+      });
+
+      it('should dispatch a CategoryPageLoadFailure action', () => {
+        expect(effects.loadCategoryPage$).toBeObservable(expected);
+      });
+    });
+	});
+
+	describe('validateFilters', () => {
+
+		it('should throw an error when an invalid range type format is given', () => {
+			expect(() => {
+				effects.validateFilters([{
+					name: 'price',
+					value: ['invalidFormat'],
+					type: DaffCategoryFilterType.Range
+				}])
+			}).toThrowError('Category range filter is in an invalid format. Should be **-**');
+		});
+
+		it('should throw an error when two filters share the same name', () => {
+			expect(() => {
+				effects.validateFilters([{
+					name: 'name',
+					value: ['value'],
+					type: DaffCategoryFilterType.Equal
+				},
+				{
+					name: 'name',
+					value: ['value2'],
+					type: DaffCategoryFilterType.Equal
+				}
+			])
+			}).toThrowError('More than one category filter of the same name exists. These should' +
+						' be combined into a single filter action with multiple values.');
+		});
+	});
+
+  describe('when CategoryPageLoadAction is triggered', () => {
+    let categoryPageLoadSuccessAction: DaffCategoryPageLoadSuccess;
+		let expected;
+		let categoryPageLoadAction;
+
+		beforeEach(() => {
+       categoryPageLoadAction = new DaffCategoryPageLoad({id: stubCategory.id});
+       categoryPageLoadSuccessAction = new DaffCategoryPageLoadSuccess({
+        category: stubCategory,
+        categoryPageConfigurationState: stubCategoryPageConfigurationState,
+        products: stubProducts
+      });
+
+      actions$ = hot('--a', { a: categoryPageLoadAction });
+		});
+
+    describe('when the call to CategoryService is successful', () => {
+      it('should dispatch a DaffCategoryPageLoadSuccess and a DaffProductGridLoadSuccess action', () => {
+				spyOn(daffCategoryDriver, 'get').and.returnValue(of({
+					category: stubCategory,
+					categoryPageConfigurationState: stubCategoryPageConfigurationState,
+					products: stubProducts
+				}));
+
+        expected = cold('--(ab)', { a: productGridLoadSuccessAction, b: categoryPageLoadSuccessAction });
+        expect(effects.loadCategoryPage$).toBeObservable(expected);
+      });
+    });
+
+    describe('when the call to CategoryService fails', () => {
+
+      beforeEach(() => {
+        const error = 'Failed to load the category';
+				const response = cold('#', {}, error);
+				spyOn(daffCategoryDriver, 'get').and.returnValue(response);
+        const categoryPageLoadFailureAction = new DaffCategoryPageLoadFailure(error);
+        expected = cold('--b', { b: categoryPageLoadFailureAction });
+      });
+
+      it('should dispatch a CategoryPageLoadFailure action', () => {
+        expect(effects.loadCategoryPage$).toBeObservable(expected);
+      });
+    });
+
+    it('should call get category with only an id', () => {
```

Comment:
I wrote this test originally, but I think it isn't quite right. It should be that it calls get category with only the request that is passed and not with any of the configurations in state.

Reply:
I think I fixed this